### ISSUE TITLE
Hypernetworks

### ISSUE THREAD

**Issue by dblunk88** [huggingface/diffusers#1140]
**Is your feature request related to a problem? Please describe.**
Support for hypernetworks since they seem to yield great results. An example can be seen here:
https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/modules/hypernetworks/hypernetwork.py

**Describe the solution you'd like**
A diffusers hypernetwork implementation. 

**Describe alternatives you've considered**
Switching back and forth between different repos, but a one-stop shop is preferred.

**Comment by WASasquatch** [comment]
This is a really cool feature, and would be awesome in diffusers.

**Comment by patrickvonplaten** [comment]
Very much agree - we should definitely add this. In general there are multiple features that come out recently that require to modify the attentions for each layer - maybe we could support all of them in one go:
- https://github.com/cloneofsimo/paint-with-words-sd 
- Hypernetworks
- am I missing one more @patil-suraj ?

**Comment by patrickvonplaten** [comment]
@patil-suraj maybe if you already have a good idea here on how to implement hypernetworks you could share here?

**Comment by aungkhant0911** [comment]
@patrickvonplaten unrelated: would be nice to have Google's Prompt2Prompt feature. https://prompt-to-prompt.github.io/

**Comment by patil-suraj** [comment]
This one also https://github.com/google/prompt-to-prompt

It would cool to have these, haven't looked deeply into it, will assign some time for it soon and reply here :)

**Comment by patrickvonplaten** [comment]
@aungkhant0911 - we're also always happy about community contributions :-)

**Comment by duongna21** [comment]
Happy to work on prompt-to-prompt.

**Comment by dblunk88** [comment]
Any updates on this and/or any help needed?

**Comment by jinwonkim93** [comment]
Hi @patrickvonplaten . Are you planning to implement only prompt-to-prompt or general method for modifying layer? I think https://github.com/cloneofsimo/paint-with-words-sd is also very cool feature and does not require much of modification.

**Comment by patrickvonplaten** [comment]
Yes in a first step the plan was to look into prompt-to-prompt. I'm a bit behind on this. Hoping to get to it by next week

**Comment by dblunk88** [comment]
any updates on this?

**Comment by patil-suraj** [comment]
Not yet. We recently added a feature to be able to easily tweak cross attention (cf #1639), which would make it easy to implement. Let us know if anyone from the community is interested in this, happy to help :)

**Comment by patrickvonplaten** [comment]
We're currently adding LoRA: https://github.com/huggingface/diffusers/pull/1884 which should be very similar to how hypernetworks are done.

**Comment by ballenvironment** [comment]
Please add this, and hypernetwork training as well!
